source/extensions/quic_listeners/quiche/envoy_quic_alarm.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "envoy/event/timer.h"
+
+#include "quiche/quic/core/quic_alarm.h"
+#include "quiche/quic/core/quic_time.h"
+#include "quiche/quic/platform/api/quic_clock.h"
+
+namespace Envoy {
+namespace Quic {
+
+// Implements QUIC interface
+// https://quiche.googlesource.com/quiche/+/refs/heads/master/quic/core/quic_alarm.h This class
+// wraps an Event::Timer object and provide interface for QUIC to interact with the timer.
+class EnvoyQuicAlarm : public quic::QuicAlarm {
+public:
+  EnvoyQuicAlarm(Event::Scheduler& scheduler, quic::QuicClock& clock,
+                 quic::QuicArenaScopedPtr<quic::QuicAlarm::Delegate> delegate);
+
+  ~EnvoyQuicAlarm() override{};
```

Comment:
nit: Can we check that IsSet() is false in destructor?

Reply:
done